### PR TITLE
Kieran/personalize analytics page with pub data

### PR DESCRIPTION
**Pull request by kieran6roberts**
## Description

Personalize the advanced analytics page `dashboard -> analytics -> advanced analytics (beta)` with blog details. Previous it just looked like analytics for Hashnode and not blog specific. This includes:

- Meta title, description, ogImage and favicon
- Header logo/title

These should match blogs for the same circumstances.

Vercel deployment of Umami analytics - https://vercel.com/hashnode-prod/hn-analytics-umami

Notes: 

- Since this analytics setup may be temporary (we might move out in the future), it was not worth implementing graphql for the pub data fetching here. GSSP fetch to an api on Main is sufficient. I checked with Bahaa who agreed.

- BEFORE MERGING TO PROD - Set `NEXT_PUBLIC_HASHNODE_ENV` in Vercel

## Testing

- add your `.env` file with the following envs:
  - NEXT_PUBLIC_HASHNODE_ENV=development
  - DATABASE_URL='Find in vercel'
  - HASH_SALT='Find in vercel'
 
- Build with `yarn build`
- Run with `yarn dev -p some other port`
- Run your hashnode servers (using the Hashnode Main `kieran/personalize-analytics-page-with-pub-data` branch).
- Now you can access the advanced analytics `/share` pages locally e.g. `/share/:umamiShareId/:UmamiWebsiteUUID`
- Umami details are on the publication model e.g. `publication.umamiShareId`.
- Test out different publications with different settings
  - Regular pub and Team pub: With/without `logo`, `dark mode logo`, `title`, `seo title`, `seo descrtiption`, `ogImage`, `favicon`.
- Open up the blog in the next tab to help with comparison. Everything should match as it does in Blogs.

### CODE REVIEW

**Comment by lakbychance** [comment]
@kieran6roberts Adding my observations here :- 
* For rogue `umamiShareId/umamiWebsiteUUID` - can we return a `404` page ?
* Unable to see author's profile photo for mobile view - 

Link to test - `http://localhost:4000/share/S7iCALZo/364340d0-926d-47f1-ba5a-042f9218d638`

Screenshot 2022-12-12 at 2 16 58 PM

Screenshot 2022-12-12 at 2 17 10 PM

File: components/common/BlogLogo.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  const [theme] = useTheme();
+  const isThemeDark = theme === 'dark';
+
+  if (!publication) {
+    return null;
+  }
+
+  return (
+    <div>
+      {publication.darkModeLogo && isThemeDark ? (
```
Comment (lakbychance) [inline components/common/BlogLogo.js]: @kieran6roberts Can you state how the order of these ternary expressions work ?  It's less readable at the moment. Also if these can be extracted into a separate function and just the prop values can be changed, that would be good ? 

P.S. - Want to understand the branching here.

File: utils/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+export const lightOrDark = color => {
```
Comment (lakbychance) [inline utils/index.js]: I didn't see the usage of this function in new code.

File: next.config.js [hunk 3]
```diff
@@ -10,6 +10,9 @@ module.exports = {
   eslint: {
     ignoreDuringBuilds: true,
   },
+  images: {
+    domains: ['cdn.hashnode.com'],
```
Comment (lakbychance) [inline next.config.js:14]: Just to make sure. We don't load any images from other domains for this Umami project right ?

File: components/common/BlogLogo.js (hunk 1 above)
Comment (kieran6roberts) [reply components/common/BlogLogo.js]: This was just taken from the blogs repo initially, but I have refactored this to hopefully be easier to understand here. Maybe then I could refactor it over there while I'm working on the blog navbar updates.

File: utils/index.js (hunk 2 above)
Comment (kieran6roberts) [reply utils/index.js]: Yep, have removed it.

File: next.config.js (hunk 3 above)
Comment (kieran6roberts) [reply next.config.js:14]: I don't believe so.

**Comment by kieran6roberts** [comment]
> @kieran6roberts Adding my observations here :-
> 
> * For rogue `umamiShareId/umamiWebsiteUUID` - can we return a `404` page ?
> * Unable to see author's profile photo for mobile view -


- Have updated to return not found when we can't get the publication using the given `umamiShareId`.
- The profile image is intentionally hidden on mobile, the same as we do for blogs. We could include the profile image (for regular publications) in the top left corner if you think that's better (I'll show an example below). Let me know what you think.

Screenshot 2022-12-12 at 10 37 28

**Review by lakbychance — APPROVED** [review]
LGTM

**Pull request merged**